### PR TITLE
fix: sorting form is not confirming by "Enter" key

### PR DESCRIPTION
**Pull request by tarashaliuk**
closes: [HRZ-3087](https://spryker.atlassian.net/browse/HRZ-3087)

[HRZ-3087]: https://spryker.atlassian.net/browse/HRZ-3087?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/picking/src/components/filters/filters.component.ts [hunk 1]
```diff
@@ -90,14 +90,14 @@ export class FiltersComponent extends LitElement {
           <button>${i18n('picking.filter.reset')}</button>
         </oryx-button>
 
-        <form @submit=${this.onSubmit}>
+        <form @submit=${this.onSubmit} id="filter-form">
```
Comment (supproduction) [inline libs/domain/picking/src/components/filters/filters.component.ts]: Why do we need form id?

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/picking/src/components/filters/filters.component.ts (hunk 1 above)
Comment (tarashaliuk) [reply libs/domain/picking/src/components/filters/filters.component.ts]: To connect form with button which is outside of the form. To make it we need 'id' and 'form' attributes

Comment (supproduction) [reply libs/domain/picking/src/components/filters/filters.component.ts]: Can we use kepress listener on this button and don't pollute html with unnecessary id's?

Comment (tarashaliuk) [reply libs/domain/picking/src/components/filters/filters.component.ts]: But the current approach is more HTML. It generates the additional `id` on the form but reduces using JS code.
@tobi-or-not-tobi what do you think?

Comment (tobi-or-not-tobi) [reply libs/domain/picking/src/components/filters/filters.component.ts]: we do indeed not like IDs, even generated. Lets try an alternative solution, let me know if you need help.

File: libs/domain/picking/src/components/filters/filters.component.ts [hunk 2]
```diff
@@ -72,6 +77,13 @@ export class FiltersComponent extends LitElement {
     this.form.requestSubmit();
   }
 
+  protected onFormKeyDown(event: KeyboardEvent): void {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      this.onApply();
```
Comment (supproduction) [inline libs/domain/picking/src/components/filters/filters.component.ts]: Please move `if (event.key === 'Enter') {event.preventDefault()} ` to apply method

File: libs/domain/picking/src/components/filters/filters.component.ts [hunk 3]
```diff
@@ -24,6 +24,11 @@ export class FiltersComponent extends LitElement {
   protected fieldRenderer = resolve(FormRenderer);
   protected pickingListService = resolve(PickingListService);
 
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.form?.removeEventListener('keydown', this.onFormKeyDown);
+  }
```
Comment (supproduction) [inline libs/domain/picking/src/components/filters/filters.component.ts]: We don't need it anymore

**Review by supproduction — CHANGES_REQUESTED** [review]

**Review by supproduction — APPROVED** [review]

**Pull request merged**